### ISSUE TITLE
Unavailable area-metric hides buildings

### ISSUE THREAD

**Issue by Richargh** [MaibornWolff/codecharta#358]
Some metrics are not available for every building. Those buildings are then not displayed in the CodeMap. That als means that some buildings can be overlooked without knowing about them. 

How should we fix this?

**Comment by alschmut** [comment]
One option were to show flat buildings with area value of 1 (= unary), but we have to make sure, that the user can see (for example currently in the detailPanel) that the areaValue of `1` is fake

**Comment by Richargh** [comment]
That's a good idea. Not sure how to visualize that it is fake though. Maybe by showing a special symbol instead of a number? Wikipedia has a list of them `https://en.wikipedia.org/wiki/List_of_mathematical_symbols` but nothing comes to mind. Or maybe there is one in fontawesome?

I also thought that we could show, next to size metric, a counter that shows how many buildings are hidden by that metric.

**Comment by NearW** [comment]
But we have 3 different visualisation parameters.
If the metric that uses the color of a building is not available, it would be easy to just add a 'error-color' to it.
If the height-metric is not available, make it flat and add an icon on top of the building surface.
If the area-metric is not available, make it 1x1 and display an icon on top as well.

**Comment by Richargh** [comment]
We have three metrics and they should be handled individually. Currently if a building has no value for a specific metric (f.ex. `sonar_code_smells`) CodeCharta dies the following:

* color metric: treat it as zero. The building becomes green. (or red if inverted)
* height metric: treat it as zero. The building is flat.
* area metric: treat it as zero. The building **disappears**.

I think the first two are obivous and not confusing. 

The last one, area, is the problem because the building just disappears. We introduced the `unary` metric just for this case. Unary is a virtual metric that assigns every building the value one. 

So the last metric is really the issue. It's a big problem from a usability perspective . First it's not obvious that buildings disappear. Second I don't have a way to use my favorite metric (`ncloc` or `rloc` which might not be calculated for all buildings) and still see all buildings.

**Comment by BridgeAR** [comment]
> I think the first two are obivous and not confusing.

I agree with the height metric. It seems fine to just keep it flat. But the color should likely be grey to indicate that the building does not use the currently selected metric.

> area metric: treat it as zero. The building disappears.

We could either use a minimum for the area of 1. That would solve the visual aspect and clicking on a building would indicate that the metric is actually 0. But this would obfuscate metrics with low values in average (especially 1). Thus, I wonder if we just want to add 1 to each metric to visualize it a tad different. This might again result in some confusion though due to non conforming aspect ratios with increased metric values. One way that would always solve it is to multiply the size by 2 and use a minimum of 1. That way the aspect ratio would always be valid and it's possible to distinguish a building without the metric (area size of 1) and a building with the metric set to 1 (area size of 2).

@RomanenkoVladimir @ce-bo this is likely related to building vanishing (while not being the only one).

**Comment by BridgeAR** [comment]
Blocked by #1609

**Comment by BridgeAR** [comment]
This is unblocked due to #3077 landing :partying_face:. Finding a concrete implementation to fix this is likely the tricky point though.

I also wondered if we just want to show an indicator next to the number of files that are not visible in a folder (just adding a sign and a number about hidden files).

**Comment by ce-bo** [comment]
I would like to address this soonish.

**Comment by CodeCharta** [comment]
This is an discussion and became a duplicate of #3780
